lib/adf_diag.py

```diff
@@ -946,7 +946,11 @@ def derive_variables(self,vars_to_derive=None,ts_dir=None):
             if var == "PRECT":
                  # PRECT can be found by simply adding PRECL and PRECC
                  # grab file names for the PRECL and PRECC files from the case ts directory
-                 constit_files = sorted(glob.glob(ts_dir+"/*PREC*"))
+                 if len(glob.glob(os.path.join(ts_dir,"*PRECC*"))) and len(glob.glob(os.path.join(ts_dir,"*PRECL*"))):
+                     constit_files=sorted(glob.glob(os.path.join(ts_dir,"*PREC*")))
+                 else:
+                     ermsg = "PRECC and PRECL were not both present; PRECT cannot be calculated."
+                     raise FileNotFoundError(ermsg)
```

Comment:
Thanks for this!  I might just change this code slightly to something like the following:

```suggestion
                 if glob.glob(os.path.join(ts_dir,"*PRECC*")) and glob.glob(os.path.join(ts_dir,"*PRECL*")):
                     constit_files=sorted(glob.glob(os.path.join(ts_dir,"*PREC*")))
                 else:
                     ermsg = "PRECC and PRECL were not both present; PRECT cannot be calculated."
                     ermsg += " Please remove PRECT from diag_var_list or find the relevant CAM files."
                     raise FileNotFoundError(ermsg)
```

Reply:
Done.